src/Chocolatey.PowerShell/CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+
+## Add `ShouldProcess` support where applicable
+
+As a _very_ brief and reductive overview, if a cmdlet is making changes to the user's machine, it should implement `ShouldProcess` support.
+This enables the built-in PowerShell functionality for `-WhatIf` and `-Confirm` parameters on the cmdlet.
+
+### Implementing `ShouldProcess`
+
+To implement `ShouldProcess` correctly, the following conditions must be met:
+
+1. The cmdlet class should be decorated with `[Cmdlet(SupportsShouldProcess = true)]` (in other words: add `SupportsShouldProcess = true` to the existing property declarations in the `Cmdlet` attribute)
+2. In any logic paths where we might be making changes to the user's machine (installing, uninstalling, modifying the registry, making changes to the user or machine environment variables, creating or deleting non-temporary files, running external applications, and so on) we need to be wrapping those code paths in an `if` check like so:
+  ```csharp
+  if (ShouldProcess("target item (path, object, env var name, description of what is being actually modified)", "description of the action to be performed"))
+  {
+	  // Code to run that does the described action.
+  }
+  ```
+
+> :memo: **Note**
+>
+> If this check takes place in a helper class rather than the main cmdlet class, remember that the `ShouldProcess` method only exists on the `PSCmdlet` base class.
+> To call `ShouldProcess` from a helper class, a `PSCmdlet` parameter can be passed to the method, and cmdlets calling into it can simply pass `this` for that parameter value: `HelperClass.MethodName(this)`
+> Then in the helper class, it can be called on the parameter, similar to this: `cmdlet.ShouldProcess(...)`
+
+Ideally these checks should take place at the deepest or most narrow point that is sensible in the code path, bypassing _only_ the code which actually makes changes to the system where possible.
+This pattern enables us to use `-WhatIf` to verify code paths when invoked from a cmdlet, without risking making any permanent changes to the testing environment.
+
+For more information on ShouldProcess, see [Everything you wanted to know about ShouldProcess](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.4).
+
+## No dependency injection
+
+In most areas of the Chocolatey CLI codebase, we use a SimpleInjector framework for dependency injection.
+This is not available in the Chocolatey.PowerShell library, as it is intentionally decoupled from the other CLI projects and should never depend on any of the other CLI projects.
+
+Additionally, since all cmdlets are initialised by the PowerShell runtime, we cannot use a dependency injection framework to make things work.
+As a result, many of the helpers will be a little more bare-bones and not make as heavy use of interfaces as other code areas in this repository.
+
+## Interop with Chocolatey CLI
```

Comment:
For completeness @gep13 could you review this section as it was added after your approval.